pulp_rpm/tests/functional/api/test_sync.py

```diff
@@ -803,69 +803,64 @@ def test_sync_advisory_incomplete_pgk_list(self):
 
         self.assertIn(error_msg, task_result["error"]["description"])
 
-    @unittest.skip("FIXME: Enable this test after #2245 is fixed")
     def test_sync_advisory_no_updated_date(self):
         """Test sync advisory with no update.
 
         1. Sync repository with advisory which has updated_date
         2. Re-sync with repo with same id and version as previous
            but missing updated_date (issued_date should be used instead).
         """
-        body = gen_rpm_remote(RPM_UNSIGNED_FIXTURE_URL)
-        remote = self.remote_api.create(body)
+        remote = self.remote_api.create(gen_rpm_remote(RPM_UNSIGNED_FIXTURE_URL))
+        self.addCleanup(self.remote_api.delete, remote.pulp_href)
 
         # sync
         repo, remote = self.do_test(remote=remote)
+        self.addCleanup(self.repo_api.delete, repo.pulp_href)
+        repository_version = repo.to_dict()["latest_version_href"]
 
-        # add remote to clean up
+        remote = self.remote_api.create(gen_rpm_remote(RPM_ADVISORY_NO_UPDATED_DATE))
         self.addCleanup(self.remote_api.delete, remote.pulp_href)
 
-        body = gen_rpm_remote(RPM_ADVISORY_NO_DATES)
-        remote = self.remote_api.create(body)
-
         # re-sync
         repo, remote = self.do_test(repo, remote)
+        repository_version_new = repo.to_dict()["latest_version_href"]
 
-        # add resources to clean up
-        self.addCleanup(self.repo_api.delete, repo.pulp_href)
-        self.addCleanup(self.remote_api.delete, remote.pulp_href)
-
-        added_advisory_date = [
-            advisory["updated_date"]
-            for advisory in get_added_content(repo.to_dict())[PULP_TYPE_ADVISORY]
-            if RPM_ADVISORY_TEST_ID in advisory["id"]
-        ]
-        removed_advisory_date = [
-            advisory["issued_date"]
-            for advisory in get_removed_content(repo.to_dict())[PULP_TYPE_ADVISORY]
-            if RPM_ADVISORY_TEST_ID in advisory["id"]
-        ]
+        # TODO: this test isn't ideal because in this case the new advisory with no "updated_date"
```

Comment:
I really wish we had a better way of unit testing things like this because making new fixtures for every scenario is problematic.